services/feed-manager-service/feed-manager-controller/src/main/java/com/thinkbiganalytics/feedmgr/rest/controller/FeedRestController.java

```diff
@@ -858,50 +859,47 @@ public DatasourceDefinitions updateDatasourceDefinitions(DatasourceDefinitions d
     @Path("/{feedId}/upload-file")
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     @Produces(MediaType.APPLICATION_JSON)
-    @ApiOperation("Uploads a file to be ingested by a feed.")
+    @ApiOperation("Uploads files to be ingested by a feed.")
     @ApiResponses({
-                      @ApiResponse(code = 200, message = "The file is ready to be ingested."),
-                      @ApiResponse(code = 500, message = "The file could not be saved.", response = RestResponseStatus.class)
+                      @ApiResponse(code = 200, message = "Files are ready to be ingested."),
+                      @ApiResponse(code = 500, message = "Files could not be saved.", response = RestResponseStatus.class)
                   })
-    public Response uploadFile(@PathParam("feedId") String feedId,
-                               @FormDataParam("file") InputStream fileInputStream,
-                               @FormDataParam("file") FormDataContentDisposition fileMetaData) throws Exception {
+    public Response uploadFile(@PathParam("feedId") String feedId, FormDataMultiPart multiPart) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

Reply:
@zhenbzha I know this isn't your issue but if you could help refactor the code to fix this, I'd appreciate it. The IOException could be changed to a javax.ws.rs.InternalServerErrorException. Any other exceptions can be handled similarly.

Reply:
sure. can help on that.

Reply:
done